climpred/classes.py

```diff
@@ -408,11 +403,13 @@ def add_uninitialized(self, xobj):
         """
         if isinstance(xobj, xr.DataArray):
             xobj = xobj.to_dataset()
-        match_initialized_dims(self.initialized, xobj)
+        match_initialized_dims(self.initialized, xobj, uninitialized=True)
         match_initialized_vars(self.initialized, xobj)
         self.uninitialized = xobj
 
-    def compute_metric(self, refname=None, metric='pearson_r', comparison='e2r'):
+    def compute_metric(
+        self, refname=None, metric='pearson_r', comparison='e2r', max_dfs=False
```

Comment:
Since I work on pandas a lot, I reserve the acronym "df" to pd.DataFrame; when it's `max_dfs`, I inferred it as like max number of dataframes. Maybe you could do `dof` instead? (scipy usees dof) https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chi2_contingency.html